source/developers-guide/shopware-5-upgrade-guide-for-developers/index.md

```diff
@@ -17,14 +17,205 @@ including minor and bugfix releases, refer to the `UPGRADE.md` file found in you
+        'write_backlog' => false,
+        'enabled' => false,
+    ],
+    ...
+],
+```
+
+To activate elasticsearch in the backend, simply change `backend.enabled` to `true` and use the `sw:es:backend:index:populate`
+command to populate the necessary indices. If you also enabled the `backend.write_log`, you can use `sw:es:backend:sync`
+periodically after that to keep the index in sync.
+
+### Changed execution model of `replace` hooks
+
+When multiple `replace` hooks exist for one method, up to 5.4 (incl.) each of these hooks were called sequentially with
+each of these hooks having the opportunity to call `executeParent()`. Hence, if this happened, the original (parent) 
+implementation got called multiple times, leading to unexpected behaviours. 
+
+In Shopware 5.5 the execution model was changed to a decorative type of implementation: If more than one `replace` hook
+exists for a function, calling `executeParent()` inside the hook will execute the next `replace` hook of function. Only
+the last `replace` hook will then call the original implementation on `executeParent()`.
+ 
+### Filesystem abstraction layer
+
+As of Shopware v5.5, the use of direct file system access methods (like e.g. `fopen()`, `file()`, `is_readable()` and many more)
+is being discouraged. Instead, we rely on the library <a href="https://github.com/thephpleague/flysystem" target="_blank">FlySystem</a>
+as a generic file system abstraction layer. This allows a lot of flexibility in regards to where files are being stored.
+E.g. Documents, ESD and Sitemap files can now also served from S3 or Google Cloud. We added support for 
+Amazon Web Services and Google Cloud Platform for the moment but this layer will help us in the future to support more 
+cloud services. 
+
+Multiple instances of this abstraction classes are available to plugins. The services with the ids `shopware.filesystem.public`
+and `shopware.filesystem.private` are available to Shopware itself and to all plugins that are active. These services are
+used to access a kind of shared "folder" in which files can be stored that might be shared between Shopware and plugins.
+The service `shopware.filesystem.public.url_generator` can be used for generating a URL to a file in the public filesystem.
+
+The `shopware.filesystem.public` file system is intended for files that need to be accessible to clients directly, e.g. 
+images, product manuals etc. while `shopware.filesystem.private` is for files that need to be available throughout Shopware
+(e.g. invoice pdfs, ESDs like software), but are not to be accessed by a browser directly without any authentication.
+
+This is a simple example of how a file can be written using the new system:
+
+```php
+$fileName = 'HelloWorld.txt';
+$contents = 'Uploaded by Shopware using FlySystem';
+
+$filesystem = $this->container->get('shopware.filesystem.private');
+if ($filesystem->has($fileName)) {
+    $filesystem->delete($fileName);
+}
+
+$filesystem->write($file, $contents);
```

Comment:
`$file` should be named `$fileName` right?

Reply:
Right